inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp

```diff
@@ -810,24 +819,34 @@ void MKLDNNGraph::Infer(MKLDNNInferRequest* request, int batch) {
 
     ENABLE_CPU_DEBUG_CAP(NodeDumper nd(config.debugCaps, infer_count));
 
-    for (int i = 0; i < graphNodes.size(); i++) {
-        if (request != nullptr) {
+#ifdef CPU_DEBUG_CAPS
+    for (int i = 0; i < constantGraphNodes.size(); i++) {
+        if (request != nullptr)
             request->ThrowIfCanceled();
-        }
 
-        PERF(graphNodes[i]);
+        ENABLE_CPU_DEBUG_CAP(nd.dumpInputBlobs(constantGraphNodes[i]));
+        ENABLE_CPU_DEBUG_CAP(nd.dumpOutputBlobs(constantGraphNodes[i]));
+    }
+#endif
+
+    std::unique_ptr<PerfHelper> perf;
+    const bool collectPerfCounters = config.collectPerfCounters;
+    for (int i = 0; i < mutableGraphNodes.size(); i++) {
+        if (request != nullptr)
+            request->ThrowIfCanceled();
+
+        if (collectPerfCounters)
+            perf = PERF(mutableGraphNodes[i]);
```

Comment:
If I got it right with this change we start including `ThrowIfCanceled()` into pc (but from the next iteration). In that case I want to understand how perf counters are affected? It also makes sense to move `perf` initialization in the very beginning of the loop.